static/app/views/organizationLayout/index.tsx

```diff
@@ -33,20 +37,57 @@ function OrganizationLayout({children}: Props) {
   // oganization is loaded before rendering children. Organization may not be
   // loaded yet when this first renders.
   const organization = useOrganization({allowNull: true});
+  const hasNavigationV2 = organization?.features.includes('navigation-sidebar-v2');
+  const App = hasNavigationV2 ? AppLayout : LegacyAppLayout;
```

Comment:
Something to watch out for: this note calls out that **"Organization may not be loaded yet when this first renders"**. 

For this experience to exist behind a feature flag, the organization has to be loaded. This could potentially cause some layout thrashing once the org loads.